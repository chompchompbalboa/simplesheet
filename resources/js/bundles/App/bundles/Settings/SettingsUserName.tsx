//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { IAppState } from '@app/state'

import { updateUser } from '@app/state/user/actions'

import SettingsLabelledInput from '@app/bundles/Settings/SettingsLabelledInput'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const UserProfileName = () => {

  const dispatch = useDispatch()

  const userId = useSelector((state: IAppState) => state.user.id)
  const userName = useSelector((state: IAppState) => state.user.name)

  const [ localUserName, setLocalUserName ] = useState(userName)

  const updateUserName = () => {
    if(userName !== localUserName) {
      dispatch(updateUser(
        userId, 
        { name: localUserName },
        'USER_UPDATE_USER_NAME_ERROR'
      ))
    }
  }

  return (
    <SettingsLabelledInput
      label="Name:"
      onBlur={() => updateUserName()}
      onChange={nextUserName => setLocalUserName(nextUserName)}
      value={localUserName}
      width="20%"/>
  )
}

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default UserProfileName
