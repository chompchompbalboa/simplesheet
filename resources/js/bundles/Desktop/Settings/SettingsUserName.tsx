//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { IAppState } from '@/state'

import { updateUser } from '@/state/user/actions'

import SettingsLabelledInput from '@desktop/Settings/SettingsLabelledInput'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SettingsUserName = () => {

  // Redux
  const dispatch = useDispatch()
  const userId = useSelector((state: IAppState) => state.user.id)
  const userName = useSelector((state: IAppState) => state.user.name)

  // State
  const [ localUserName, setLocalUserName ] = useState(userName)

  // Update User Name
  const updateUserName = () => {
    if(userName !== localUserName) {
      dispatch(updateUser(
        userId, 
        { name: localUserName },
        'USER_UPDATE_USER_NAME_ERROR'
      ))
    }
  }

  return (
    <SettingsLabelledInput
      label="Name:"
      onBlur={() => updateUserName()}
      onChange={nextUserName => setLocalUserName(nextUserName)}
      value={localUserName}
      width="30%"/>
  )
}

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SettingsUserName
