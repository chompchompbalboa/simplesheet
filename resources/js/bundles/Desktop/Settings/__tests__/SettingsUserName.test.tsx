//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import axiosMock from 'axios'

import { renderWithRedux } from '@/testing/library'
import { appStateFactory, IAppStateFactoryInput } from '@/testing/mocks/appState'

import SettingsUserName from '@desktop/Settings/SettingsUserName'

//-----------------------------------------------------------------------------
// Setup
//-----------------------------------------------------------------------------
const {
  allFiles,
  allFileIds,
  allSheetsFromDatabase
} = appStateFactory({} as IAppStateFactoryInput)

const fileId = allFileIds[0]
const sheetId = allFiles[fileId].typeId
const sheetFromDatabase = allSheetsFromDatabase[sheetId]

// @ts-ignore
axiosMock.get.mockResolvedValue({ data: sheetFromDatabase })

console.warn = jest.fn()
console.error = jest.fn()

//-----------------------------------------------------------------------------
// Tests
//-----------------------------------------------------------------------------
describe('SettingsUserName', () => {
  
  it("renders without crashing", async () => {
    const container = renderWithRedux(<SettingsUserName />)
    expect(container).toMatchSnapshot()
  })
  
  it("display a 'Name:' input", async () => {
    const { queryByLabelText } = renderWithRedux(<SettingsUserName />)
    expect(queryByLabelText('Name:')).toBeTruthy()
  })

})
