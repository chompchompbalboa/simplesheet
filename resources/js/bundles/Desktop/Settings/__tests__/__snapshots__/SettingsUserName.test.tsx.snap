// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsUserName renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-bottom: 0.5rem;
  padding: 0.25rem 0.125rem;
  width: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgb(50,50,50);
  border-bottom: 1px dashed rgb(200,200,200);
}

.c1 {
  display: inline;
  font-weight: bold;
  font-size: 0.85rem;
  white-space: nowrap;
}

.c2 {
  width: 100%;
  text-align: right;
  border: none;
  outline: none;
  background-color: transparent;
  font-family: inherit;
  font-size: 0.8rem;
  color: rgb(50,50,50);
}

<body>
    <div>
      <div
        class="c0"
      >
        <label
          class="c1"
        >
          Name:
        </label>
        <input
          class="c2"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AykKC kUoGTD"
    >
      <label
        class="sc-AykKD Tocjq"
      >
        Name:
      </label>
      <input
        class="sc-AykKE jyHzqb"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
